### PR TITLE
docs: remove note on disk space for caching

### PR DESCRIPTION
**Pull request by ltalirz**
The new repository implementation includes automatic de-duplication of identical files.
Re-running a cached calculation should therefore not result in copies of the results stored
in the repository, and in no increase in disk space usage besides what is needed for storing
metadata for the new calculation nodes, data nodes & links in the database.

### CODE REVIEW

**Comment by sphuber** [comment]
I wouldn't remove this, or at the very least just adjust the text. Even though content in the file repository is now deduplicated, this is just for the `psql_dos` implementation. It is not guaranteed for other storage backends. Also, the content in the database for the `psql_dos` is not deduplicated. If you have nodes with a lot of content, that will be cloned in postgres.

**Comment by ltalirz** [comment]
Thanks for the comment @sphuber 

Well, the gist of the sentence is certainly no longer correct and needs to change.

In my view, it would also be fine to delete since the duplication of metadata on the DB level is unlikely to substantially impact disk usage - and there is value to users not spending time worrying about things that are unlikely to impact them.

Given that this is under "topics", however, where people go to learn how things work, I'm also fine with providing a more detailed explanation here. 
Let me give it a try.

File: docs/source/topics/provenance/caching.rst [hunk 1]
```diff
@@ -159,7 +159,8 @@ Limitations and Guidelines
    While AiiDA's hashes include the version of the Python package containing the calculation/data classes, it cannot detect cases where the underlying Python code was changed without increasing the version number.
    Another scenario that can lead to an erroneous cache hit is if the parser and calculation are not implemented as part of the same Python package, because the calculation nodes store only the name, but not the version of the used parser.
 
-#. Note that while caching saves unnecessary computations, it does not save disk space: the output nodes of the cached calculation are full copies of the original outputs.
+#. While caching saves unnecessary computations, it does not directly prevent duplication of data: the cached calculation and its output nodes are duplicated.
+   In practice, however, AiiDA's file repository implementation will detect that any files associated with these nodes are already present and simply point to those, reducing duplication to metadata stored at the database level.
```
[the remark this answers is not in the document]
Comment (ltalirz) [reply docs/source/topics/provenance/caching.rst]: thanks, I wanted to phrase it slightly differently but I've incorporated your suggestions

File: docs/source/topics/provenance/caching.rst [hunk 2]
```diff
@@ -159,7 +159,9 @@ Limitations and Guidelines
    While AiiDA's hashes include the version of the Python package containing the calculation/data classes, it cannot detect cases where the underlying Python code was changed without increasing the version number.
    Another scenario that can lead to an erroneous cache hit is if the parser and calculation are not implemented as part of the same Python package, because the calculation nodes store only the name, but not the version of the used parser.
 
-#. Note that while caching saves unnecessary computations, it does not save disk space: the output nodes of the cached calculation are full copies of the original outputs.
+#. While caching saves unnecessary computations, it does lead result in duplication of the cached calculation and its output nodes in the provenance graph.
+   At the same time, AiiDA's default disk-objectstore storage backend comes with automatic de-duplication at the object level.
```
Comment (sphuber) [inline docs/source/topics/provenance/caching.rst:163]:
```suggestion
#. While caching saves unnecessary computations, it does not necessarily save space as the cached calculation and its output nodes are duplicated in the provenance graph.
   However, AiiDA's default disk-objectstore storage backend comes with automatic de-duplication at the object level.
```

**Review by sphuber — CHANGES_REQUESTED** [review]

**Pull request merged**